website/cue/reference/components/sources/statsd.cue

```diff
@@ -124,6 +124,17 @@ components: sources: statsd: {
 				[metric data model](\(urls.vector_metric)) page for more info.
 				"""
 		}
+		tags: {
+			title: "Tags"
+			body: """
+				The StatsD protocol removes any colon character found after the tag key:value
+				separator. For instance, the tagset "tagA:val::ue", is parsed by StatsD as
+				"tagA=value".
```

Comment:
I think this is a little unclear. IIUC the "protocol" does remove the colon characters, only the server implementation. On the other hand, _our_ implementation doesn't remove them, so this mention confuses more than it clarifies.

Reply:
This section was added based on this discussion:
https://github.com/vectordotdev/vector/pull/15617#discussion_r1052584902

I agree that the term protocol is misused here, and that in it's current state the section isn't a great addition.

It seems like that suggestion was because it wasn't clear from

```
if chunk contains more than one :, it will ignore everything after the second :.
```

, that the statsd server behavior means we don't need to handle a case of multiple colon chars.

It seems like our options are to either

a. Abandon the is section in the docs altogether.
b. Adapt it to something like

```
The StatsD server will remove any colon character found after the tag key:value separator. For instance, the tag key-value pair "tagA:val::ue", is parsed by StatsD as "tagA=value" . Thus Vector only looks for the first colon character in a tag key-value pair.
```

What do you think?

Independent of the external doc, I can also improve the comment in the code to make it more clear that this

```
            // note: the behavior of statsd if more than one colon is found (which would presumably
            // be part of the tag value), is to remove any additional colons from the tag value.
```

means

```
            // hence the usage of `find()` matching the first colon character is sufficient.

Reply:
I don't see this first part having much value to a user. The tags coming from statsd already have the colons stripped, and we aren't doing anything to change that, nor are we stripping colons from incoming tags if they had them. IOW the current behavior _is_ the least surprising behavior and explaining that sounds like it's going to introduce more confusion. The second part is worth keeping to document that the trailing colon is parsed as an empty string and not a bare tag, which we do distinguish internally.

Similar with the code comment, our behavior is simply to preserve the value of incoming tags. Unless the documented protocol says we _SHOULD_ be stripping colons from tags, it doesn't really matter of statsd has or hasn't stripped the colons, vector is simply preserving the value as it has received them. OTOH it might be worth documenting that we diverge from the behavior of statsd (kinda sorta).

Reply:
I attempted to update the doc and comment in accordance with that and our discussion offline, in https://github.com/vectordotdev/vector/pull/15617/commits/72f5d098b0a4ca03e579b8e077d89a694a00199c.